lib/puppet/provider/certificate_file/certificate_file.rb

```diff
@@ -0,0 +1,24 @@
+Puppet::Type.type(:certificate_file).provide(:openssl) do
+  commands :openssl => 'openssl'
+
+  def source_cert
+    if valid?(resource[:source_cert])
+      resource[:source_cert]
+    else
+      raise ArgumentError, "Source certificate is not a valid x509 certificate"
+    end
+  end
+
+  def valid?(certificate)
+    File.exist?(certificate)
+    pem_format?(certificate)
+  end
```

Comment:
Other things we could attempt to do is pass in the private key to this resource and validate that they have the same modulo to ensure they will work together. If we passed in the CA certificate that is supposed to be supplied with the certificate we could additionally verify they work together.